server/routes/notification.js

```diff
@@ -220,6 +221,24 @@ router.post("/price", async (req, res) => {
 
     await transaction.commit();
 
+    //Hashmap for keeping track who are the users that we already sent an event to
+    let notifiedUsers = {};
+
+    //We iterate the userLists which contains all the affected users of the price change.
+    //We check here if the user is NOT in the notifiedUsers hashmap, if so we emit an event
+    //and add the userId to the hashmap so that if the users userId comes up again, we dont 
+    //send them another emit again.
+    userLists.forEach((userList) => {
+      let stringUserId = `${userList.userId}`;
+
+      if (userSockets.connections[stringUserId]) {
+        if (!notifiedUsers[stringUserId]) {
+          userSockets.connections[stringUserId][0].emit("new-notifications");
```

Comment:
I would rewrite this as `userSockets.connection[stringUserId].forEach( (socket) => socket.emit("new-notification") )` so we notify the user on all instances of our site that they have open.
Also, it looks like you could send the notification info with this emit, and then we would cut down on a later backend request and db query. Just a thought though!